scripts/run_tests.sh

```
@@ -32,21 +32,25 @@ function display_result {
   fi
 }
 
-flake8 --max-complexity 10 --count
-display_result $? 1 "Code style check"
 
-py.test --cov=app --cov-report xml $@ $1
-display_result $? 2 "Unit tests"
+if [ -z "$EQ_FUNCTIONAL_TEST_SUITES" ]; then
+    export EQ_FUNCTIONAL_TEST_SUITES="core"
+fi
 
-# Run front end tests
-npm config set python python2.7
+if [ "$EQ_FUNCTIONAL_TEST_SUITES" = "core" ]; then
```

If a unit test fails then the other builds will run all of the functional tests? surely this breaks the fail fast rule?

---

I guess you still get the fail fast indication from the other build, the disadvantage is that you don't free up your travis resource I guess. I am just playing at the moment to see how much time it saves us to see if the cost is worth the benefit

---

The results are:

Running all tests on both matrix:
https://travis-ci.org/ONSdigital/eq-survey-runner/builds/184486318
https://travis-ci.org/ONSdigital/eq-survey-runner/builds/184489761

Running all tests on core:
https://travis-ci.org/ONSdigital/eq-survey-runner/builds/184491190

---

We could fail the census branch if the core branch fails to free up resources, that said we can't do that until travis get their act together - https://github.com/travis-ci/travis-ci/issues/2062

---

i can't see much difference in the build times.

---

I've updated my comment above to include the times